prowler/providers/aws/services/ec2/ec2_service.py

```diff
@@ -407,6 +411,46 @@ def __get_ebs_encryption_settings__(self, regional_client):
                 f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
 
+    def __get_snapshot_block_public_access_state__(self, regional_client):
+        try:
+            snapshot_in_region = False
+            for snapshot in self.snapshots:
+                if snapshot.region == regional_client.region:
+                    snapshot_in_region = True
+            self.ebs_block_public_access_snapshots_state.append(
+                EbsSnapshotBlockPublicAccess(
+                    status=regional_client.get_snapshot_block_public_access_state()[
+                        "State"
+                    ],
+                    snapshots=snapshot_in_region,
+                    region=regional_client.region,
+                )
+            )
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+    def __get_instance_metadata_defaults__(self, regional_client):
+        try:
+            instances_in_region = False
+            for instance in self.instances:
```

Comment:
You need to add another `try/except` clause within the loop to handle exceptions there and not to exit the loop execution.